utils/helpers/schedule/periodNumberAt.ts

```diff
@@ -14,7 +14,7 @@ export default function periodNumberAt(date: Date = new Date()): number {
   // period 1.
   const calculatedPeriod =
     Math.floor(
-      differenceInMinutes(zonedDate, getTodaySetToPeriodTime(1)) / 50,
+      differenceInMinutes(zonedDate, setDateToPeriodTime(zonedDate, 1)) / 50,
```

Comment:
Why is this implemented? The changing of timezones won't be a thing that can happen in mysk, would it?

Reply:
There’s something you have to understand about timezones, it’s that it comes to bite you in the ass when you least expect. Remember all the problems with printing receipts for SK Shopping? That was timezones! Vercel was on UTC and we assumed it was on UTC+7 because we were idiots.

Here’s an actual explanation:

### Server-side Time
When `new Date()` is called on the server (i.e. Vercel), it uses the timezone from the `TZ` environment variable. The timezone on Vercel is UTC. For example, if it is 7 PM in Bangkok, it’s 12 PM on the server.

This is a problem because we expect time-based calculations to use UTC+7. For example, when we say school starts at 8:30 AM UTC+7, we expect the condition to kick in at 8:30 AM UTC+7. **However, because the server is on UTC, it kicks in at 8:30 AM UTC instead, which is 3:30 PM UTC+7.**

### Client-side Time
`new Date()` called in the browser uses the user’s local timezone. If the user is in Bangkok, the timezone is set to UTC+7. If the user is in New York, the timezone is set to UTC-4.

Problems arise when the user is not located in UTC+7 time. Using the school start time example again, **if the user is in New York, school would start 11 hours after it’s supposed to, as the condition uses New York time instead of Bangkok.**

### Reconciliation
- We put `TZ=Etc/UTC` in the environment variables file so our timezone matches that of Vercel.
  - This helps with catching timezone bugs without having to deploy to Vercel.
- We use `toZonedTime` to convert all dates into the school’s timezone (defined by `NEXT_PUBLIC_SCHOOL_TZ`) before using it in time-based calculations.